editoast/src/views/stdcm_search_environment.rs

```diff
@@ -304,4 +389,124 @@ pub mod tests {
+
+        let pool = app.db_pool();
+
+        let (
+            infra1,
+            timetable1,
+            work_schedule_group1,
+            temporary_speed_limit_group1,
+            electrical_profile_set1,
+        ) = stdcm_search_env_fixtures(&mut pool.get_ok()).await;
+
+        let env1 = StdcmSearchEnvironment::changeset()
+            .infra_id(infra1.id)
+            .electrical_profile_set_id(Some(electrical_profile_set1.id))
+            .work_schedule_group_id(Some(work_schedule_group1.id))
+            .temporary_speed_limit_group_id(Some(temporary_speed_limit_group1.id))
+            .timetable_id(timetable1.id)
+            .search_window_begin(Utc.with_ymd_and_hms(2024, 1, 2, 0, 0, 0).unwrap())
+            .search_window_end(Utc.with_ymd_and_hms(2024, 1, 15, 0, 0, 0).unwrap())
+            .enabled_from(Utc.with_ymd_and_hms(2024, 1, 1, 0, 0, 0).unwrap())
+            .enabled_until(Utc.with_ymd_and_hms(2024, 1, 1, 23, 59, 59).unwrap());
+
+        let env1_tweaked = env1
+            .clone()
+            .enabled_from(Utc.with_ymd_and_hms(2023, 1, 1, 0, 0, 0).unwrap())
+            .enabled_until(Utc.with_ymd_and_hms(2023, 1, 1, 23, 59, 59).unwrap());
+
+        let (
+            infra2,
+            timetable2,
+            work_schedule_group2,
+            temporary_speed_limit_group2,
+            electrical_profile_set2,
+        ) = stdcm_search_env_fixtures(&mut pool.get_ok()).await;
+
+        let env2 = StdcmSearchEnvironment::changeset()
+            .infra_id(infra2.id)
+            .electrical_profile_set_id(Some(electrical_profile_set2.id))
+            .work_schedule_group_id(Some(work_schedule_group2.id))
+            .temporary_speed_limit_group_id(Some(temporary_speed_limit_group2.id))
+            .timetable_id(timetable2.id)
+            .search_window_begin(Utc.with_ymd_and_hms(2024, 1, 2, 0, 0, 0).unwrap())
+            .search_window_end(Utc.with_ymd_and_hms(2024, 1, 15, 0, 0, 0).unwrap())
+            .enabled_from(Utc.with_ymd_and_hms(2024, 1, 1, 0, 0, 0).unwrap())
+            .enabled_until(Utc.with_ymd_and_hms(2024, 1, 1, 23, 59, 59).unwrap());
+
+        for env in [env1, env2, env1_tweaked] {
+            env.create(&mut pool.get_ok())
+                .await
+                .expect("Failed to create stdcm search environment");
+        }
+
+        let request = app.get("/stdcm/search_environment/list");
+
+        // WHEN
+        let stdcm_search_envs = app
+            .fetch(request)
+            .assert_status(StatusCode::OK)
+            .json_into::<Vec<StdcmSearchEnvironment>>();
+
+        // THEN
+        assert_eq!(stdcm_search_envs.len(), 3);
+        assert_eq!(
+            stdcm_search_envs
+                .iter()
+                .map(|env| env.infra_id)
+                .sorted()
+                .collect_vec(),
+            [infra1.id, infra1.id, infra2.id]
+                .into_iter()
+                .sorted()
+                .collect_vec()
+        );
+    }
```

Comment:
```suggestion
        assert_eq!(
            stdcm_search_envs
                .iter()
                .map(|env| env.infra_id)
                .collect(),
            HashSet::from([infra1.id, infra1.id, infra2.id])
        );
    }
```

no need to assert the size beforehand if you're already comparing collections (and it makes the failure message a bit better by printing the actual collections instead of just the size mismatch)

Reply:
Ah indded my first assert was uneeded ^^

Not sure about the second change though, collect into a hashset would lose count information, wouldn't it?

Reply:
Theres only ids in the set, which are guaranteed to be unique. So no deduplication can happen while processing the response ids. Or am I missing something?

Reply:
The same infra id is present twice ^^

Reply:
Dammit read too fast, it's infra IDs... The best imo would be to assert on env IDs directly, but since our tests are all flaky that can't be a hard equality. How about?

```suggestion
        assert_eq!(
        	HashSet::from([env1.id, env2.id, env3.id])
        		.difference(
        			&stdcm_search_envs
                		.iter()
                		.map(|env| env.id)
                		.collect()
                ),
            HashSet::default()
        );
    }
```

Reply:
Thanks! But what do you mean they're all flaky? Is there a chance envs don't actually get created?

Reply:
They operate on a postgres launched externally, so there may be data already in the db when the test starts. Like if you load a dump in your db, and start the test, the dump data will be queried. (But data created in tests are never persisted after the it exits though.)

There might already be some stdcm environments defined before you create the ones you want to list. 

This problem never occurs in the CI since a new db is created for each run.